src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/PolicyFunctions.kt

```diff
@@ -32,18 +32,27 @@ fun <T : ILicenceCondition> conditionChanges(current: List<T>, other: List<T>) =
   amendedConditions(current, other)
 )
 
+/**
+ * This will return a list of all changes that need to be addressed as part of migrating a licence from one version of the policy to another.
+ */
 fun <T : Any> licencePolicyChanges(
-  presentConditions: List<AdditionalCondition>,
+  licence: Licence,
   previousConditions: List<IAdditionalCondition<T>>,
   currentConditions: List<IAdditionalCondition<T>>,
   allReplacements: List<Replacements>
 ): List<LicenceConditionChanges> {
-  val conditionsToCheck = currentConditions.filter { current ->
-    presentConditions.any { present -> current.code == present.code }
+  val conditionsToCheck = previousConditions.filter { policyCondition ->
+    licence.additionalLicenceConditions.any { licenceCondition ->
+      licenceCondition.code == policyCondition.code && licence.version != licenceCondition.version
+    }
   }
-  val changes = conditionChanges(conditionsToCheck, previousConditions, allReplacements)
+  val changes = conditionChanges(conditionsToCheck, currentConditions, allReplacements)
```

Comment:
previous and current conditions were swapped around previously, I hadn't got test coverage here so have added some